### PR TITLE
Update ImageBuilder components

### PR DESCRIPTION
**Pull request by stephenmsachs**
*Issue #, if available:*

*Description of changes:*
* Remove redundant imagebuilder-components.yaml
* Fix README links
* Use ImageBuilder WebDownload instead of `curl`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: recipes/pcs/hpc_ready_ami/assets/components/optimize-performance.yaml [hunk 1]
```diff
@@ -39,13 +39,16 @@ Resources:
...
+              - name: DownloadCommon
+                action: WebDownload
                 inputs:
-                  commands:
-                    - set -e
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh -o common.sh'
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/optimize-performance.sh -o optimize-performance.sh'
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh
+                    destination: common.sh
+              - name: DownloadAgent
```
Comment (javydekoning) [inline recipes/pcs/hpc_ready_ami/assets/components/optimize-performance.yaml]: Incorrectly named. Should say something like "DownloadOptimizeScript"

File: recipes/pcs/hpc_ready_ami/assets/components/optimize-performance.yaml [hunk 2]
```diff
@@ -39,13 +39,16 @@ Resources:
...
+                action: WebDownload
+                inputs:
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/optimize-performance.sh
+                    destination: optimize-performance.sh
               - name: InstallAgent
```
Comment (javydekoning) [inline recipes/pcs/hpc_ready_ami/assets/components/optimize-performance.yaml]: See previous comment.

File: recipes/pcs/hpc_ready_ami/assets/components/install-efs-utils.yaml [hunk 3]
```diff
@@ -39,13 +39,16 @@ Resources:
...
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh -o common.sh'
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/install-efs-utils.sh -o install-efs-utils.sh'
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh
+                    destination: common.sh
+              - name: DownloadInstaller
+                action: WebDownload
+                inputs:
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/install-efs-utils.sh
+                    destination: install-efs-utils.sh
               - name: InstallAgent
```
Comment (javydekoning) [inline recipes/pcs/hpc_ready_ami/assets/components/install-efs-utils.yaml]: Incorrect name

File: recipes/pcs/hpc_ready_ami/assets/components/install-lustre.yaml [hunk 4]
```diff
@@ -40,13 +40,16 @@ Resources:
...
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh -o common.sh'
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/install-lustre.sh -o install-lustre.sh'
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh
+                    destination: common.sh
+              - name: DownloadAgent
+                action: WebDownload
+                inputs:
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/install-lustre.sh
+                    destination: install-lustre.sh
               - name: InstallAgent
```
Comment (javydekoning) [inline recipes/pcs/hpc_ready_ami/assets/components/install-lustre.yaml]: Incorrect name, Lustre is Kernel Module

File: recipes/pcs/hpc_ready_ami/assets/components/update-os.yaml [hunk 5]
```diff
@@ -39,13 +39,16 @@ Resources:
...
+              - name: DownloadCommon
+                action: WebDownload
                 inputs:
-                  commands:
-                    - set -e
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh -o common.sh'
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/update-os.sh -o update-os.sh'
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh
+                    destination: common.sh
+              - name: DownloadAgent
```
Comment (javydekoning) [inline recipes/pcs/hpc_ready_ami/assets/components/update-os.yaml]: Why not use:

```sh
  - name: UpdateLinux
    action: UpdateOS
```

Comment (stephenmsachs) [reply recipes/pcs/hpc_ready_ami/assets/components/update-os.yaml]: Does not work for [all OSs](https://docs.aws.amazon.com/imagebuilder/latest/userguide/toe-action-modules.html#action-modules-updateos) we want to support: "UpdateOS doesn't support Amazon Linux 2023 (AL2023)."

**Comment by stephenmsachs** [comment]
Thank you @javydekoning for your review.

**Pull request merged**